### PR TITLE
ARC: boards: nsim: hs5x/hs6x: setup apertures to be runnable on HAPS

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
Setup peripheral and NOC apertures to make hs5x and hs6x configurations (both UP and SMP) runnable on real HW (HAPS).

PS: later we can try to implement it in a more generic way, though I'm not fully sure what is the best way to do that, as such setup is quite SoC-specific. Till then let's keep it in nsim* board platform code.

### CODE REVIEW

**Review by abrodkin — DISMISSED** [review]

File: boards/arc/nsim/haps_arcv3_init.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+#define ARC_CLN_MST_NOC_0_3_ADDR	2564
+#define ARC_CLN_MST_NOC_0_3_SIZE	2565
+
+#define ARC_CLN_MST_NOC_0_4_ADDR	2566
+#define ARC_CLN_MST_NOC_0_4_SIZE	2567
+
+#define ARC_CONNECT_CMD_CLN_PER0_BASE	2688
+#define ARC_CONNECT_CMD_CLN_PER0_SIZE	2689
+
+#define AUX_CLN_ADDR			0x640
```
Comment (ruuddw) [inline boards/arc/nsim/haps_arcv3_init.c:28]: Why mixing hex and decimal in this list? I'd prefer hex for all.

File: boards/arc/nsim/haps_arcv3_init.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	z_arc_v2_aux_reg_write(AUX_CLN_ADDR, ARC_CONNECT_CMD_CLN_PER0_BASE);
+	z_arc_v2_aux_reg_write(AUX_CLN_DATA, 0xF00);
+	z_arc_v2_aux_reg_write(AUX_CLN_ADDR, ARC_CONNECT_CMD_CLN_PER0_SIZE);
+	z_arc_v2_aux_reg_write(AUX_CLN_DATA, 1);
+
+	z_arc_v2_aux_reg_write(AUX_CLN_ADDR, ARC_CLN_MST_NOC_0_0_ADDR);
+	z_arc_v2_aux_reg_write(AUX_CLN_DATA, (DT_REG_ADDR(DT_CHOSEN(zephyr_sram)) >> 20));
+	z_arc_v2_aux_reg_write(AUX_CLN_ADDR, ARC_CLN_MST_NOC_0_0_SIZE);
+	z_arc_v2_aux_reg_write(AUX_CLN_DATA, (DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) / (1024 * 1024)));
```
Comment (ruuddw) [inline boards/arc/nsim/haps_arcv3_init.c:44]: Another small inconsistency. ADDR is manipulated by shifting, size by a divide. Result is the same, why not use the same type of expression for both?

**Review by ruuddw — DISMISSED** [review]
Minor style suggestions.
Board-level initialization is fine I think, things like 'sram size' (external memory in this case, I assume) is board specific.

A more general approach would be to have 'drivers' (couldn't the ARC_CONNECT programming be done with the current driver for that?) for the controls and initialize/configure them all from dts entries.

File: boards/arc/nsim/haps_arcv3_init.c (hunk 1 above)
Comment (evgeniy-paltsev) [reply boards/arc/nsim/haps_arcv3_init.c:28]: It's different address spaces (AUX address space and cluster address space which is accessible via two AUX regs) - so even in documentation they are defined differently (AUX regs in hex, cluster regs in decimal).

Though I can recalculate everything to hex, if you prefer so.

File: boards/arc/nsim/haps_arcv3_init.c (hunk 2 above)
Comment (evgeniy-paltsev) [reply boards/arc/nsim/haps_arcv3_init.c:44]: Fixed

**Review by abrodkin — APPROVED** [review]

File: boards/arc/nsim/haps_arcv3_init.c (hunk 1 above)
Comment (ruuddw) [reply boards/arc/nsim/haps_arcv3_init.c:28]: if this is consistent with the documentation, then let's leave it like this.

**Review by ruuddw — APPROVED** [review]

**Pull request merged**